### ISSUE TITLE
Ability to render GroundOverlays on top of map content such as buildings

### ISSUE THREAD

**Issue by zalexbag** [googlemaps/android-maps-compose#526]
**Is your feature request related to a problem? Please describe.**
GroundOverlays always render below map content (specifically buildings that are enabled via `isBuildingEnabled = true` in `MapProperties` for instance) and there isn't any way to make GroundOverlays be rendered on top (i.e. their z index is always below everything else on the map).

 **Describe the solution you'd like**
Add an API that offers the ability to for GroundOverlays to be rendered above other map content, and especially for buildings as that is needed for my team's use case where we want buildings to be drawn below our GroundOverlay image.

 **Describe alternatives you've considered**
We tried working with the various shape APIs (Circle, Polygon, and Polyline) to see if any of those rendered above map content and found that none of those do either, though those also don't offer the same functionality we need which only GroundOverlay offers anyways.

 **Additional context**
Below you can see examples of how buildings are always rendered on top of GroundOverlay, Circle, and Polygon:

| GroundOverlay | Circle | Polygon |
| ------ | ------ | ------ |
|   ![fpom-groundoverlay](https://github.com/googlemaps/android-maps-compose/assets/136486013/2dfde318-d184-496b-be52-d550471e26dc)    |    ![fpom-circle](https://github.com/googlemaps/android-maps-compose/assets/136486013/e8e4d7f9-38d7-4e45-87a2-00f0407a9c92)    |  ![fpom-polygon](https://github.com/googlemaps/android-maps-compose/assets/136486013/b5df7e18-e03a-47d7-9f44-94f62cbf5dae)   |

**Comment by wangela** [comment]
If you would like to upvote the priority of this issue, please comment below or react on the original post above with :+1: so we can see what is popular when we triage.
--------

@zalexbag Thank you for opening this issue. 🙏
Please check out these other resources that might help you get to a resolution in the meantime:

- [Check the issue tracker](https://developers.google.com/maps/support/#issue_tracker) - bugs and feature requests for Google Maps Platform APIs and SDKs
- [Open a support case](https://developers.google.com/maps/support#contact-maps-support) - Get 1:1 support in Cloud Console.
- [Discord](https://discord.gg/fHTDmZe9En) - chat with other developers
- [StackOverflow](http://stackoverflow.com/questions/tagged/google-maps) - use the `google-maps` tag


_This is an automated message, feel free to ignore._

**Comment by zalexbag** [comment]
@lucanicoletti yes I did test changing the `zIndex` to a few different non-zero values including `Float.MAX_VALUE` and buildings still render on top of the GroundOverlay.

**Comment by zalexbag** [comment]
Oh and one other point of context to add to that, in the Maps Android SDK documentation for Shapes in the [Z-index section](https://developers.google.com/maps/documentation/android-sdk/shapes#z-index) they state the following:

> The z-index specifies the stack order of this shape, relative to other overlays (other shapes, ground overlays and tile overlays) on the map. An overlay with a high z-index is drawn above overlays with lower z-indexes. Two overlays with the same z-index are drawn in an arbitrary order.
>
> Note that markers are always drawn above other overlays, regardless of the z-index of the other overlays.

This is referring to shapes, but I would infer the behavior is the same for other overlay types and in this case for GroundOverlay. So the way I see it, the `zIndex` is meant to layer with regards to other overlays but has no effect when non-overlay elements are displayed like buildings or (as they mentioned) map markers.

**Comment by Hugo-Persson** [comment]
@zalexbag did you find some solution?

**Comment by zalexbag** [comment]
@Hugo-Persson no, I did not find a solution yet 🫤 .

**Comment by zalexbag** [comment]
Reporting back here just to mention one other important point that I forgot to mention originally, this behavior does not align with Google Maps behavior on iOS and Web. Both of those platforms show the overlay on top of 3D buildings. So to make sure there is feature parity across all platforms, this issue should be prioritized for Android.

**Comment by umutm** [comment]
@zalexbag I'm having the same issue on Android. Did you manage to find a workaround for it?

**Comment by zalexbag** [comment]
Hey @umutm , no I haven't found a workaround. This issue has been put on the back burner for my team until Google is able to release a fix. That's very interesting that you managed to get it to work temporarily. There must be some kind of discrepancy as you suggest. I'll be interested to hear if you manage to find a workaround!

**Comment by kikoso** [comment]
Hi @zalexbag .  This is a fixed behavior of the underlying Google Maps Android SDK. `GroundOverlay` objects are explicitly designed to be drawn flat against the earth's surface, meaning 3D map features like buildings, extruded structures, and POI labels will inherently render on top of them. The Maps Compose library cannot override the base SDK's internal Z-ordering rules for map primitives.
